esphome/components/at581x/at581x.cpp

```diff
@@ -0,0 +1,213 @@
+#include "at581x.h"
+
+/* Select gain for AT581X (3dB per step for level1, 6dB per step for level 2), high value = small gain. (p12) */
+const uint8_t GAIN_ADDR_TABLE[] = {0x5c, 0x63};
+const uint8_t GAIN5C_TABLE[] = {0x08, 0x18, 0x28, 0x38, 0x48, 0x58, 0x68, 0x78, 0x88, 0x98, 0xa8, 0xb8, 0xc8};
+const uint8_t GAIN63_TABLE[] = {0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06};
+const uint8_t GAIN61_VALUE = 0xCA;  // 0xC0 | 0x02 (freq present) | 0x08 (gain present)
+
+/*!< Power consumption configuration table (p12). */
+const uint8_t POWER_TABLE[] = {48, 56, 63, 70, 77, 91, 105, 115, 40, 44, 47, 51, 54, 61, 68, 78};
+const uint8_t POWER67_TABLE[] = {0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7};
+const uint8_t POWER68_TABLE[] = {0x8, 0x8, 0x8, 0x8, 0x8, 0x8, 0x8, 0x8,
+                                 24,  24,  24,  24,  24,  24,  24,  24};  // See Page 12, shift by 3 bits
+
+/*!< Frequency Configuration table (p14/15 of datasheet). */
+const uint8_t FREQ_ADDR = 0x61;
+const uint16_t FREQ_TABLE[] = {5696, 5715, 5730, 5748, 5765, 5784, 5800, 5819, 5836, 5851, 5869, 5888};
+const uint8_t FREQ5F_TABLE[] = {0x40, 0x41, 0x42, 0x43, 0x44, 0x45, 0x46, 0x47, 0x40, 0x41, 0x42, 0x43};
+const uint8_t FREQ60_TABLE[] = {0x9d, 0x9d, 0x9d, 0x9d, 0x9d, 0x9d, 0x9d, 0x9d, 0x9e, 0x9e, 0x9e, 0x9e};
+
+/*!< Value for RF and analog modules switch (p10). */
+const uint8_t RF_OFF_TABLE[] = {0x46, 0xaa, 0x50};
+const uint8_t RF_ON_TABLE[] = {0x45, 0x55, 0xA0};
+const uint8_t RF_REG_ADDR[] = {0x5d, 0x62, 0x51};
+
+/*!< Registers of Lighting delay time. Unit: ms, min 2s (p8) */
+const uint8_t HIGH_LEVEL_DELAY_CONTROL_ADDR = 0x41; /*!< Time_flag_out_ctrl 0x01 */
+const uint8_t HIGH_LEVEL_DELAY_VALUE_ADDR = 0x42;   /*!< Time_flag_out_1 Bit<7:0> */
+
+const uint8_t RESET_ADDR = 0x00;
+
+/*!< Sensing distance address */
+const uint8_t SIGNAL_DETECTION_THRESHOLD_ADDR_LO = 0x10;
+const uint8_t SIGNAL_DETECTION_THRESHOLD_ADDR_HI = 0x11;
+
+/*!< Bit field value for power registers */
+const uint8_t POWER_THRESHOLD_ADDR_HI = 0x68;
+const uint8_t POWER_THRESHOLD_ADDR_LO = 0x67;
+const uint8_t PWR_WORK_TIME_EN = 8;     // Reg 0x67
+const uint8_t PWR_BURST_TIME_EN = 32;   // Reg 0x68
+const uint8_t PWR_THRESH_EN = 64;       // Reg 0x68
+const uint8_t PWR_THRESH_VAL_EN = 128;  // Reg 0x67
+
+/*!< Times */
+const uint8_t TRIGGER_BASE_TIME_ADDR = 0x3D;  // 4 bytes, so up to 0x40
+const uint8_t PROTECT_TIME_ADDR = 0x4E;       // 2 bytes, up to 0x4F
+const uint8_t TRIGGER_KEEP_TIME_ADDR = 0x42;  // 4 bytes, so up to 0x45
+const uint8_t TIME41_VALUE = 1;
+const uint8_t SELF_CHECK_TIME_ADDR = 0x38;  // 2 bytes, up to 0x39
+
+namespace esphome {
+namespace at581x {
+
+static const char *const TAG = "at581x";
+
+AT581XComponent::AT581XComponent() {}
+
+bool AT581XComponent::i2c_write_reg(uint8_t addr, uint8_t data) {
+  return write_register(addr, &data, 1) == esphome::i2c::NO_ERROR;
```

Comment:
All member function calls and variables should be prefixed with `this->` please
```suggestion
  return this->write_register(addr, &data, 1) == esphome::i2c::NO_ERROR;
```